### PR TITLE
HOTT-846: Lint fixes to spec controllers

### PR DESCRIPTION
**Pull request by jebw**
Lint fixes to controllers I'll be changing

### Jira link

[HOTT-846](https://transformuk.atlassian.net/browse/HOTT-846)

### What?

I have added/removed/altered:

- [x] Linted spec files I will be changing 

### Why?

I am doing this because:

- to simplify the PR with the changes

### CODE REVIEW

File: app/models/commodity.rb [hunk 1]
```diff
@@ -14,22 +14,24 @@ class Commodity < GoodsNomenclature
   has_many :ancestors, class_name: 'Commodity'
 
   delegate :goods_nomenclature_item_id, :display_short_code, to: :heading, prefix: true
-  alias :short_code :goods_nomenclature_item_id
+  alias_method :short_code, :goods_nomenclature_item_id
 
   def substring=(substring)
+    # rubocop:disable Naming/MemoizedInstanceVariableName
```
Comment (alexdesi) [inline app/models/commodity.rb:20]: really? 
I am wondering how we should Memoize in this case?
sometimes Rubocop is annoying

File: spec/controllers/commodities_controller_spec.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 describe CommoditiesController, type: :controller do
   describe 'GET to #show' do
-    context 'with UK site' do
+    context 'with XI site' do
```
Comment (alexdesi) [inline spec/controllers/commodities_controller_spec.rb:5]: good catch :)

File: spec/controllers/commodities_controller_spec.rb [hunk 3]
```diff
@@ -65,8 +64,7 @@
         end
 
         it 'redirects to actual version of the commodity page' do
-          expect(response.status).to eq 302
-          expect(response.location).to eq commodity_url(id: commodity_id.first(10))
+          expect(response).to redirect_to commodity_url(id: commodity_id.first(10))
```
Comment (alexdesi) [inline spec/controllers/commodities_controller_spec.rb:67]: `expect(response.status).to eq 302` seemed a genuine test, should we keep it?

File: spec/controllers/commodities_controller_spec.rb [hunk 4]
```diff
@@ -88,16 +86,15 @@
 
         it 'redirects to heading page (strips exceeding commodity id characters)' do
           TradeTariffFrontend::ServiceChooser.service_choice = nil
-          expect(response.status).to eq 302
-          expect(response.location).to eq heading_url(id: commodity_id.first(4))
+          expect(response).to redirect_to heading_url(id: commodity_id.first(4))
```
Comment (alexdesi) [inline spec/controllers/commodities_controller_spec.rb:89]: same here

File: spec/controllers/commodities_controller_spec.rb [hunk 5]
```diff
@@ -109,8 +106,7 @@
 
         it 'redirects to actual version of the commodity page' do
           TradeTariffFrontend::ServiceChooser.service_choice = nil
-          expect(response.status).to eq 302
-          expect(response.location).to eq commodity_url(id: commodity_id.first(10))
+          expect(response).to redirect_to commodity_url(id: commodity_id.first(10))
```
Comment (alexdesi) [inline spec/controllers/commodities_controller_spec.rb:109]: and here

**Review by alexdesi — APPROVED** [review]
It looks good!
the only thing I am not sure about is the removing of the 302 HTTP response tests (redirect)

File: spec/controllers/commodities_controller_spec.rb (hunk 3 above)
Comment (jebw) [reply spec/controllers/commodities_controller_spec.rb:67]: My conclusion was we're probably not fussed about the particular status code, just that its in the 3xx range

That matcher chains through to the rails one, which checks a `3xx` status code, and checks the `response.location` ?

https://github.com/rails/rails/blob/83217025a171593547d1268651b446d3533e2019/actionpack/lib/action_dispatch/testing/assertions/response.rb#L53

Comment (alexdesi) [reply spec/controllers/commodities_controller_spec.rb:67]: I got your point, yes it sounds good to me! 👍

File: spec/controllers/commodities_controller_spec.rb (hunk 4 above)
Comment (alexdesi) [reply spec/controllers/commodities_controller_spec.rb:89]: we've agreed to keep it (see the comment before)

File: spec/controllers/commodities_controller_spec.rb (hunk 5 above)
Comment (alexdesi) [reply spec/controllers/commodities_controller_spec.rb:109]: as above

File: app/models/commodity.rb (hunk 1 above)
Comment (jebw) [reply app/models/commodity.rb:20]: Yeh, this left me stumped - hence giving up and using the `'rubocop:disable`

**Pull request merged**